### DISCUSSION TITLE
[Feature] within an album, ability to display images *not* grouped by month

### DISCUSSION THREAD

**Discussion by quitchat** [immich-app/immich#13845] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

It would be nice to have the option to display images just like how windows displays "extra large icons", rather than group by month (ie. having each month in its own row).

This is one of my albums...
![4](https://github.com/user-attachments/assets/3431663f-377a-4ff2-ae63-cb73a96111fa)

It would be nice to just display them without grouping by month, something like this..
![5](https://github.com/user-attachments/assets/6690f2ec-dfab-4019-9716-76f4f7ea3379)

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by nicokaiser** [comment]
Screen space especially on mobile devices is very limited. The app considers this and has beautiful album and timeline views. However in some cases it might be necessary to use the web view on a mobile browser, such as when an Album is shared via link.

- Add a toggle to the Album options: "Group by date" (for compatibility this can be enabled by default)
- When disabled:
	- Images in the album view are not grouped by date
	- Scubbable scrollbar is not displayed

This might be very helpful to create beautiful shared albums which can be viewed on mobile devices using the shared link.

**Comment by vvzvlad** [comment]
I support, very much want such a feature. when every day one photo, it is very difficult to leaf through, instead of a bunch of photos on the line only one. In the mobile app it is perfectly realized

**Comment by slavanap** [comment]
Mobile is also affected (esp search results).
Different granularity (month, year) would be good, too

**Comment by ggrafu** [comment]
This essentially puts immich on pair with Google Photos(where such grid view is default) making adoption easier. Love to see it

**Comment by doncezart** [comment]
+1
I've been yearning for this for months, and it's the only thing that keeps me from switching to immich for all of my media
This feature is already present in the albums page, where you can switch off grouping by date
Would be nice to also have it for the general timeline, not only inside albums

**Comment by ChrisPrefect** [comment]
I just installed immich and saw that it is basically useless for me because it groups all my photos. This makes most of my screen white because most days have one or two photos and it is very cumbersom to scroll trough a lof of photos. I don't want any grouping. Exactly like in the iOS native photo app. Just a continous grid of all my photos. Dates have no meaning for my photos. Groupying by them is as useful as grouping by the hash value of the photo itself. I don't know why this cannot be disabled? There is an option for grouping in the ios app, but "automatic" does not help and "no grouping" is missing. 

This is marked as only "web", but it is the same issue in the iOS app under "Fotos" and all the Albums.

**Comment by jurjendijkstra** [comment]
I agree. Sorting by date is one thing but grouping is too ugly, in my opinion. Please don't group photos in an album. 

Now I will need to keep Google Photo's to have presentable albums, that's a shame.

(I am using Chrome and Android app)

**Comment by ChrisPrefect** [comment]
This request is 1.5 years old and it is a very basic feature to implement. Do the developers check these discussions? Or do we need to open up an actuall issue?

My next try is to download the iOS project and ask claude code to add a "no grouping" option so I can actually use the app for myself. 

(Would it be allowed to publish the forked ios app to the app store?)

**Comment by gingerlime** [comment]
I also bumped into it after importing my library, but sadly immich didn't pass "the wife test" because of this. For the main photo view the timeline is amazing, but inside albums it breaks. We have albums with few photos spanning several dates/years. The view becomes sparse with lots of empty space, so you need to scroll more. Switching to "Group by month" kinda helps on Mobile, though not really. This option doesn't exist on Web at all.

Would be great to be able to select how galleries/albums/memories (on this day) are displayed.

<img width="2352" height="1476" alt="CleanShot 2026-02-10 at 09 10 18" src="https://github.com/user-attachments/assets/a1f20c9e-c484-42a7-9259-2a43d6f49c82" />

**Reply by gingerlime** [reply]
By the way, the Album list itself has a nice feature to toggle grouping. Something similar *inside* albums would be perfect.

<img width="1876" height="1042" alt="CleanShot 2026-02-10 at 09 14 57" src="https://github.com/user-attachments/assets/ad3c37e3-3aec-4db1-93fb-1133c1253abc" />

**Comment by jurjendijkstra** [comment]
I think albums have 2 purposes:  (1) to make a good looking collection, and (2) to share with friends or family. 
Immich fails at point 1, and the nature of self-hosting makes (2) challenging too. 

So for sharing I will duplicate the things I want to share to Google Photo and/or Flickr. I see no other solution.

**Reply by nicokaiser** [reply]
(2) depends on the kind of self-hosting: when running on a VPS, or using some kind of reverse proxy, (i.e. public instance) sharing album links works perfectly fine.

But I agree on (1): especially for "mixed" albums with images from many different dates, always grouping by date does not make any sense.

It is to be seen if it's easy to switch off: I could imagine the thumbnail view does this automatically, and building a thumbnail view without date slider and grouping requires quite some effort beyond "just don't display the date".

**Reply by gingerlime** [reply]
100%. I self-host it, but have an external domain name, so sharing is not a problem technically. I wouldn't want to share something that doesn't look easy to browse though ;-)

I don't know the internals of Immich, but intuitively any kind of grouping is usually *more* complicated than no grouping at all. So removing the grouping from Albums (by default, always, or as a toggle) should in theory be easier than applying it.

**Reply by jurjendijkstra** [reply]
Now I have come to accept that Immich albums are not for good-looking presentations, but only for collecting my things for myself, I actually begin to find this grouping by date convenient :-) 

For example: I periodically make images of something that's in a process of change. The grouping by date really helps :-)

**Reply by gingerlime** [reply]
I agree that it helps in some cases, but not all. In this particular scenario it's not helpful at all.

**Reply by jurjendijkstra** [reply]
Of course. It just needs a little switch for grouping options, by default "no grouping" (but still sorting by date taken).

**Comment by DrSapling** [comment]
Does anybody know, whether this feature is in the works or the devs at least noticed it?

I want to expand a little on the idea, by additionally allowing for grouping scale change, like: day, week, month, year and none. (The last one would disable grouping)

**Comment by jurjendijkstra** [comment]
I am in piece with it now, I actually started to like the "break by date" thing. The only one who is going to see the albums is me. I do not give anybody else access to my server. In the rare cases that I want to share an collection I select the photos and upload them to flickr or google photos where my folks can access it. My immich is for me alone and the break by date is actually useful, not beautiful but thats alright

**Comment by ChrisPrefect** [comment]
Guys, we have OpenAI Codex and Claude Opus now. Just tell your agent „fork this project and remove the grouping of the fotos in the album view“. Build and push to your phone. Done.

I added like 20 new features to the iOS app and Webserver like this. Works perfectly. Github issues and change request are a thing of the past, just let your agent do it directly.

**Reply by gingerlime** [reply]
I thought about it. For the web it seems straightforward but how do I publish a forked iOS app? have no experience with it but using TestFlight isn’t really an option nor paying Apple to publish an app. Would love to find out if there are better ways I wasn’t aware of.

**Reply by ChrisPrefect** [reply]
You need a mac. Install Xcode and the Codex desktop app. Tell it to download this github repo and modify the iOS project. In xcode load the modified project and select your phone as target. Activate developer mode on your iphone before.

Codex can explain everything to you if you don’t know what to do.

**Reply by gingerlime** [reply]
Thanks. And to push it to my wife’s phone too I guess. Somehow feels like more tinkering than I’m comfortable with, but I’ll look into it.

**Comment by nicokaiser** [comment]
IMO this is a basic feature which is really necessary. Actually it is two features:

- Photo sorting in albums (allowing photos in albums to be sorted by other criteria than date – with "manual sorting" probably being the hardest to implement
- Optionally disabling grouping for the photo grid. Currently all instances of the photo grid (timeline, all filter views, etc.) are grouped by day. The grid must be modified to also allow no grouping at all. However this might come with some serious performance issues, since e.g. disabling grouping in the timeline view would result in a huuuuge single photo grid for thousands of photos, which might very well crash every browser.

Both are not trivial and require thorough consideration, and surely not something which can easily (and sustainably) done by an AI.

**Reply by gingerlime** [reply]
Regarding 2 UI without grouping, this is already handled on the main view without issues, and possibly in other views.

**Reply by ChrisPrefect** [reply]
Yes this is a basic feature. An the fact that this feature request was created two years ago with no response from the developers tells me they don’t care. That’s why I implemented it myself. It works perfectly. No more grouping in all the albums. On iOS and the Webview. No performance issues at all. It even feels much faster now.

Codex and Opus can handle features like this without issues and probably better than a developer himself. I implemented it sustainably by letting the LLM create a patch documentation of each new feature so it can apply each patch again after a new version is forked again.

I also added features like uploading of photos from my hidden album on iOS to the locked folder on the server (why was this not already implemented? Makes no sense), adding a filter button that automatically hides all photos of documents, labels, devices, screenshots etc. to only show real photos, upload fotos from me and my wife to the same album but having buttons to filter by uploader, removing the clutter like cloud-icons and review-albums and ken-burns-album-headers etc.

If you want a feature forking and using Codex is faster and you can personalize it perfectly to your needs.

**Reply by gingerlime** [reply]
would you consider making a PR for this feature specifically? not sure it it will change the fact but if the implementation is simple enough, maybe it’ll help make a case for it?

**Reply by ChrisPrefect** [reply]
Making a PR and having to defend it is way more work than just asking Codex to do it for your own version 😅

I might release my version with all the new features to the Appstore maybe. As far as I have checked the license this is possible.

**Reply by gingerlime** [reply]
yes I totally understand but at least it moves this one step forward?

if you release a fork I’d be the first to try it haha

**Reply by ChrisPrefect** [reply]
Sure 😎 https://github.com/ChrisPrefect/immich
First menu under settings lists all the changes and each one can be enabled or disabled. Some also require the updated web server.

But realisticly it is easier for you to just ask Codex to implement the specific features you want on your own fork.

**Reply by gingerlime** [reply]
Thank you. The biggest hurdle for me is the ios app.

**Comment by rrweller** [comment]
I agree with this feature request, as there are times where I have albums that are very sparsely populated in terms of date, resulting in just a couple images per month. It would be nice to disable grouping entirely in cases like this.
